lib/ssl/src/ssl_handshake.erl

```diff
@@ -2152,17 +2152,9 @@ do_digitally_signed(Version, Msg, HashAlgo, {#'RSAPrivateKey'{} = Key,
                                              #'RSASSA-PSS-params'{}}, SignAlgo) when ?TLS_GTE(Version, ?TLS_1_2) ->
     Options = signature_options(SignAlgo, HashAlgo),
     public_key:sign(Msg, HashAlgo, Key, Options);
-do_digitally_signed(Version, {digest, Digest}, _HashAlgo, #'RSAPrivateKey'{} = Key, rsa) when ?TLS_LTE(Version, ?TLS_1_1) ->
-    public_key:encrypt_private(Digest, Key,
-			       [{rsa_pad, rsa_pkcs1_padding}]);
-do_digitally_signed(Version, {digest, Digest}, _,
-                    #{algorithm := rsa} = Engine, rsa) when ?TLS_LTE(Version, ?TLS_1_1) ->
-    crypto:private_encrypt(rsa, Digest, maps:remove(algorithm, Engine),
-                           rsa_pkcs1_padding);
-do_digitally_signed(_, Msg, HashAlgo, #{algorithm := Alg} = Engine, SignAlgo) ->
-    Options = signature_options(SignAlgo, HashAlgo),
-    crypto:sign(Alg, HashAlgo, Msg, maps:remove(algorithm, Engine), Options);
-do_digitally_signed(Version, {digest, _} = Msg , HashAlgo, Key, _) when ?TLS_LTE(Version,?TLS_1_1) ->
+do_digitally_signed(Version, {digest, Digest}, _HashAlgo, Key, rsa) when ?TLS_LTE(Version, ?TLS_1_1) ->
+    public_key:encrypt_private(Digest, Key);
```

Comment:
You need to preserve the padding option here!

Reply:
If you refer to the old line 2160; I'm not calling `crypto:private_encrypt` directly, but  `public_key:encrypt_private` which applies `default_options([])` that adds  `[{rsa_padding, rsa_pkcs1_padding}];`. Is it too hidden?

Reply:
Ah, that should be ok then. No problem just looking at patch diffs it was not obvious :)